sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlockBlobClient.java

```diff
@@ -306,6 +313,8 @@ public void stageBlock(String base64BlockId, InputStream data, long length) {
     public Response<Void> stageBlockWithResponse(String base64BlockId, InputStream data, long length, byte[] contentMd5,
         String leaseId, Duration timeout, Context context) {
         Objects.requireNonNull(data);
+        data.mark(Integer.MAX_VALUE);
```

Comment:
Why do we mark `Integer.MAX_VALUE`? If we call `reset` after this the read position of the stream will be `Integer.MAX_VALUE`.

Reply:
The value passed to mark is the expiry limit. It says after reading N bytes invalidate the mark and throw on a call to reset. I don't think think we ever want our mark to expire. The mark is set at the position of the stream when mark is called, so the intention here is that we'll be resetting the same position that the stream was at when it was handed to us.